### DISCUSSION TITLE
Saleor + NextJS - do I need additional backend?

### DISCUSSION THREAD

**Discussion by qZanity** [saleor/saleor#13351] [Q&A]
What is the best practice with Saleor, is it best to have a third-party backend?

For example, what if saleor.io can't do something that I want to add as a feature, or add some customization. Is it best to use for a seperate backend, like Django for example. so basically will have NextJS communicating with Django and Saleor,

**Comment by djkato** [comment]
As per docs, they recommend to extend Saleors functionality via Saleor Apps: [https://docs.saleor.io/docs/3.x/developer/extending/overview](https://docs.saleor.io/docs/3.x/developer/extending/overview)

Saleor does have a CMS functionality too, all products /categories etc can have custom metatada, so if it's really that unique that saleor can't help with it it might be easier to have 2 backends imo. This isn't advice from a seasoned saleor user though, so maybe the "proper way" can vary ^^

**Comment by timuric** [comment]
Hi @qZanity we encourage to use micro services approach where graphql is used for communication. Could you provide more details about your use case? In many cases nextjs would be enough to support extensibility as a backend service
